### ISSUE TITLE
Use `_lastUpdated` in pagination when possible

### ISSUE THREAD

**Issue by codyebberson** [medplum/medplum#5005]
The current implementation of search pagination can get slow when using the `_offset` parameter with values 1m+.  Behind the scenes, the `_offset` parameter is translated to a Postgres `LIMIT` expression, which is known to be slow on large tables.

The traditional recommendation to this problem is to use "cursors". Not SQL level cursors -- logical cursors that can take advantage of IDs or whatnot.  Due to the open ended nature of FHIR search, we cannot use these logical cursors in call cases.  But we can sometimes.

When possible (frequently), we should instead use the `_lastUpdated` parameter for pagination.  `_lastUpdated` is a stable sort, server assigned, monotonically increasing, etc.  `_lastUpdated` is already our most commonly used sort parameter.  This will work for the "no sort parameter" case as well.

**Comment by rahul1** [comment]
`SearchControl` would need to be modified somehow as well., since I think it uses offset logic for its page selector

**Comment by codyebberson** [comment]
Yes.  If we take this idea all the way, it would break the ability to jump to arbitrary pages.  I suspect that would be an unpopular change, but it's also a pretty common evolutionary terminal node ☹️

**Comment by codyebberson** [comment]
Fixed in https://github.com/medplum/medplum/pull/5098 and https://github.com/medplum/medplum/pull/5102

> `SearchControl` would need to be modified somehow as well., since I think it uses offset logic for its page selector

After looking into this, we decided to leave `<SearchControl>` alone for now.
* The `_cursor` pagination is weird when it comes to the "previous" page
* We looked at storing "previous" page state inside `<SearchControl>`, but that gets weird for permalinks

We now have a server config setting for a maximum `_offset` value.  It is currently disabled.  When we do enable it, the value will probably be relatively high (10k-100k), which is unlikely to affect most human use of `<SearchControl>`

The `_cursor` implementation is versioned, so we can always tweak it and add more functionality in the future as need arises.
